From: Jiri Bohac <jbohac@suse.cz>
Subject:    REVERT: ipv6: don't flush routes when setting loopback down
Patch-mainline: no, temporary fix

Revert upstream commit 29ba5fed1bbd09c2cba890798c8f9eaab251401d.
It causes IPv6 connections to hang instead of returning an error when
IPv6 is disabled.

See ongoing discussion at http://www.spinics.net/lists/netdev/msg153093.html
seeking for a proper fix.

Signed-off-by: Jiri Bohac <jbohac@suse.cz>

diff --git a/net/ipv6/addrconf.c b/net/ipv6/addrconf.c
index 93b7a93..848b355 100644
--- a/net/ipv6/addrconf.c
+++ b/net/ipv6/addrconf.c
@@ -2669,9 +2669,7 @@ static int addrconf_ifdown(struct net_device *dev, int how)
 
 	ASSERT_RTNL();
 
+	rt6_ifdown(net, dev);
-	/* Flush routes if device is being removed or it is not loopback */
-	if (how || !(dev->flags & IFF_LOOPBACK))
-		rt6_ifdown(net, dev);
 
 	idev = __in6_dev_get(dev);
	if (idev == NULL)
